firmware/src/touch.c

```diff
@@ -155,16 +155,12 @@ bool touch_sensor_ok(unsigned i)
 
 const uint16_t *touch_raw()
 {
-    static uint16_t readout[36];
-    uint16_t buf[36];
+    static uint16_t readout[36] = {0};
 
     for (int i = 0; i < 3; i++) {
-        sensor_ok[i] = mpr121_raw(MPR121_BASE_ADDR + i, buf + i * 12, 12);
+        sensor_ok[i] = mpr121_raw(MPR121_BASE_ADDR + i, readout + i * 12, 12);
```

Comment:
mpr121_raw's i2c operation takes time and will potentially call a callback function while waiting.
So don't use the readout as working buffer, use a private buffer and when mpr121 operation finishes, update the readout.
That's what I did.
So please revert this part.

Reply:
This part is reverted, also thank you for your suggestion!